src/hdf5_back_gen.py

```diff
@@ -106,405 +386,338 @@ def __init__(self, canon):
-jlen = col_sizes_[table][j] / itemsize;
-{t.cpp} x;
-for (unsigned int k = 0; k < jlen; ++k) {{
-    x[VLRead<{t.sub[1].cpp}, {t.sub[1].db}>(buf + offset + itemsize*k)] = *reinterpret_cast<{t.sub[2].cpp}*>(buf + offset + itemsize*k + CYCLUS_SHA1_SIZE);
-}}
-{NO_CLOSE}
-""".strip()
-
-MAP_STRING_STRING_READER = """
-hid_t field_type = H5Tget_member_type(tb_type, j);
-hid_t item_type = H5Tget_super(field_type);
-hid_t key_type = H5Tget_member_type(item_type, 0);
-size_t nullpos;
-hsize_t fieldlen;
-H5Tget_array_dims2(field_type, &fieldlen);
-unsigned int itemsize = col_sizes_[table][j] / fieldlen;
-unsigned int keylen = H5Tget_size(key_type);
-unsigned int vallen = itemsize - keylen;
-{t.cpp} x;
-for (unsigned int k = 0; k < fieldlen; ++k) {{
-    {t.sub[1].cpp} key = {t.sub[1].cpp}(buf + offset + itemsize*k, keylen);
-    nullpos = key.find('\\0');
-    if (nullpos != {t.sub[1].cpp}::npos)
-        key.resize(nullpos);
-    {t.sub[2].cpp} val = {t.sub[2].cpp}(buf + offset + itemsize*k + keylen, vallen);
-    nullpos = val.find('\\0');
-    if (nullpos != {t.sub[2].cpp}::npos)
-        val.resize(nullpos);
-    x[key] = val;
-}}
-{H5TCLOSE_MULTI}
-""".strip()
-
-MAP_STRING_VL_STRING_READER = """
-hid_t field_type = H5Tget_member_type(tb_type, j);
-size_t nullpos;
-hsize_t fieldlen;
-H5Tget_array_dims2(field_type, &fieldlen);
-unsigned int itemsize = col_sizes_[table][j] / fieldlen;
-unsigned int keylen = itemsize - CYCLUS_SHA1_SIZE;
-{t.cpp} x;
-for (unsigned int k = 0; k < fieldlen; ++k) {{
-    {t.sub[1].cpp} key = {t.sub[1].cpp}(buf + offset + itemsize*k, keylen);
-    nullpos = key.find('\\0');
-    if (nullpos != {t.sub[1].cpp}::npos)
-        key.resize(nullpos);
-    x[key] = VLRead<{t.sub[2].cpp}, {t.sub[2].db}>(buf + offset + itemsize*k + keylen);
-}}
-{H5TCLOSE}
-""".strip()
-
-MAP_VL_STRING_STRING_READER = """
-hid_t field_type = H5Tget_member_type(tb_type, j);
-size_t nullpos;
-hsize_t fieldlen;
-H5Tget_array_dims2(field_type, &fieldlen);
-unsigned int itemsize = col_sizes_[table][j] / fieldlen;
-unsigned int vallen = itemsize - CYCLUS_SHA1_SIZE;
-{t.cpp} x;
-for (unsigned int k = 0; k < fieldlen; ++k) {{
-    {t.sub[2].cpp} val = {t.sub[2].cpp}(buf + offset + itemsize*k + CYCLUS_SHA1_SIZE, vallen);
-    nullpos = val.find('\\0');
-    if (nullpos != {t.sub[2].cpp}::npos)
-        val.resize(nullpos);
-    x[VLRead<{t.sub[1].cpp}, {t.sub[1].db}>(buf + offset + itemsize*k)] = val;
-}}
-{H5TCLOSE}
-""".strip()
-
-MAP_VL_STRING_VL_STRING_READER = """
-hid_t field_type = H5Tget_member_type(tb_type, j);
-size_t nullpos;
-hsize_t fieldlen;
-H5Tget_array_dims2(field_type, &fieldlen);
-unsigned int itemsize = 2*CYCLUS_SHA1_SIZE;
-{t.cpp} x;
-for(unsigned int k = 0; k < fieldlen; ++k) {{
-    x[VLRead<{t.sub[1].cpp}, {t.sub[1].db}>(buf + offset + itemsize*k)] = VLRead<{t.sub[2].cpp}, {t.sub[2].db}>(buf + offset + itemsize*k + CYCLUS_SHA1_SIZE);
-}}
-{H5TCLOSE}
+def normal_close(t):
+    """
```

Comment:
Docstrings could be a little more descriptive, such as "This function represents..."